tests/pds/pds_ds_rbac_test.go

```diff
@@ -0,0 +1,243 @@
+						log.FailOnError(err, "Failed during restore.")
+						restoredDeployment, err = restoreClient.Components.DataServiceDeployment.GetDeployment(restoredModel.GetDeploymentId())
+						resDeployments[ds] = restoredDeployment
+						log.FailOnError(err, fmt.Sprintf("Failed while fetching the restore data service instance: %v", restoredModel.GetClusterResourceName()))
+						deploymentsToBeCleaned = append(deploymentsToBeCleaned, restoredDeployment)
+						log.InfoD("Restored successfully. Details: Deployment- %v, Status - %v", restoredModel.GetClusterResourceName(), restoredModel.GetStatus())
+					}
+				})
+
+				Step("Scale up the restored deployments on ns2", func() {
+					log.InfoD("Starting to scale up the restore deployment")
+					for ds, resDep := range resDeployments {
+						log.InfoD("Scaling up DataService %v ", ds.Name)
+
+						dataServiceDefaultAppConfigID, err = controlPlane.GetAppConfTemplate(tenantID, ds.Name)
+						log.FailOnError(err, "Error while getting app configuration template")
+						dash.VerifyFatal(dataServiceDefaultAppConfigID != "", true, "Validating dataServiceDefaultAppConfigID")
+
+						dataServiceDefaultResourceTemplateID, err = controlPlane.GetResourceTemplate(tenantID, ds.Name)
+						log.FailOnError(err, "Error while getting resource setting template")
+						dash.VerifyFatal(dataServiceDefaultAppConfigID != "", true, "Validating dataServiceDefaultAppConfigID")
+
+						updatedDeployment, err := pdslib.UpdateDataServices(deployment.GetId(),
+							dataServiceDefaultAppConfigID, deployment.GetImageId(),
+							int32(ds.ScaleReplicas), dataServiceDefaultResourceTemplateID, ns2)
+						log.FailOnError(err, "Error while updating dataservices")
+
+						err = dsTest.ValidateDataServiceDeployment(updatedDeployment, ns2)
+						log.FailOnError(err, "Error while validating data service deployment")
+
+						_, _, config, err := pdslib.ValidateDataServiceVolumes(updatedDeployment, *resDep.Name, dataServiceDefaultResourceTemplateID, storageTemplateID, ns2)
+						log.FailOnError(err, "error on ValidateDataServiceVolumes method")
+						dash.VerifyFatal(int32(ds.ScaleReplicas), config.Spec.Nodes, "Validating replicas after scaling up of dataservice")
+					}
+				})
+				//ToDo : Add workload generation for restored-deps with RBAC roles on ns2
+
+				Step("Delete Deployments", func() {
+					CleanupDeployments(deploymentsToBeCleaned)
+					CleanupServiceIdentitiesAndIamRoles(siToBeCleaned, iamRolesToBeCleaned, actorId)
+					customParams.SetParamsForServiceIdentityTest(params, false)
+				})
+
+			}
+
+		})
+		JustAfterEach(func() {
+			defer EndTorpedoTest()
+		})
+	})
+})
+
+func DeployandValidateDataServicesWithSiAndTls(ds dataservice.PDSDataService, namespaceName string, namespaceid, projectID string, resourceTemplateID string, appConfigID string, dsVersion string, dsImage string, dsID string) (*pds.ModelsDeployment, map[string][]string, map[string][]string, error) {
```

Comment:
move this func to pds_common.go

Reply:
Cannot move it, as unable to access few variables like "storagetemplateid" and few other vars.